### PR TITLE
Add getUserEmail function

### PR DESCRIPTION
**Pull request by adiaaida**
When uploading perf results to benchview, we want to supply the email
address of the user who initiated the run. Return the github email
address if accessible, and if not, return <username>@github.login, so
that we don't collide with potential actual @github.com email addresses.

### CODE REVIEW

**Comment by adiaaida** [comment]
CC: @jorive @dpodder

**Comment by adiaaida** [comment]
@mmitche did I do something wrong with the PR? it's failing to checkout the library, and I don't know why.

**Comment by mmitche** [comment]
@adiaaida Push to upstream for this. rather than your own fork

**Comment by adiaaida** [comment]
@dotnet-bot test CI Tests

File: tests/pipeline/pipeline-tests.groovy [hunk 1]
```diff
@@ -132,6 +132,24 @@ stage ('Run Tests') {
...
+            "getUserEmail - GitHub PR, no email": {
+                withEnv(['ghprbPullAuthorLogin=baz', 'ghprbGhRepository=foo/bar']) {
+                    assert getUserEmail() == 'baz@github.login' : "Exped getUserEmail would return baz@github.login"
+                }
+            },
+
+            // Testing non-PR getUserEmail is tough as it returns different values based on the cause of the run.
+            // You can't mock causes in here without opening up untrusted APIs either.
+            "getUserEmail - non mocked behavior" : {
+                assert getUserEmail() != null : "Expected getUserEmail would return valid value."
```
Comment (dpodder) [inline tests/pipeline/pipeline-tests.groovy]: I think we should test
```groovy
assert getUserEmail() : "Expected getUserEmail would return valid value."
```
or perhaps
```groovy
assert !isNullOrEmpty(getUserEmail()) : "Expected getUserEmail would return valid value."
```
instead. The null check still passes if `getUserEmail()` returns emptystring, but `assert` treats emptystring as false.

**Pull request closed without merging**